pkg/compactor/compactor.go

```diff
@@ -302,6 +303,10 @@ func newCompactor(
 			Name: "cortex_compactor_tenants_processing_failed",
 			Help: "Number of tenants failed processing during the current compaction run. Reset to 0 when compactor is idle.",
 		}),
+		compactionRunInterval: promauto.With(registerer).NewGauge(prometheus.GaugeOpts{
```

Comment:
Would this be a good candidate for a [GaugeFunc](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#GaugeFunc) metric?

Reply:
Good question. I supposed it would be if we currently, or plan to, support live reloading of this config value (and others obviously) on the compactor. If not, would GaugeFunc get us anything over the way I'm doing it here?

Reply:
No big preference, but I would set the metric value here in `newCompactor()` and not at the end of `starting()`. If I'm not missing anything, metrics can be scraped while starting as well.

Reply:
No preference here either, I don't think a GaugeFunc would buy you anything beyond the config reload thing you mentioned. My thinking was really just to remove the call to set the value of the gauge which seemed a bit far away from the place the metric was defined :+1: